plugins/tor/tor_

```diff
@@ -0,0 +1,491 @@
+
+def simplify(cn):
+    """Simplify country name"""
+    cn = cn.replace(' ', '_')
+    cn = cn.replace("'", '_')
+    cn = cn.split(',', 1)[0]
+    return cn
+
+
+class ConnectionError(Exception):
+    """Error connecting to the controller"""
+
+
+class AuthError(Exception):
+    """Error authenticating to the controller"""
+
+
+def authenticate(controller):
+    try:
+        controller.authenticate()
+        return
+    except stem.connection.MissingPassword:
+        pass
+
+    try:
+        password = os.environ['torpassword']
+    except KeyError:
+        raise AuthError("Please configure the 'torpassword' "
+                        "environment variable")
+
+    try:
+        controller.authenticate(password=password)
+    except stem.connection.PasswordAuthFailed:
+        print("Authentication failed (incorrect password)")
+
+
+def gen_controller():
+    connect_method = os.environ.get('connectmethod', default_torconnectmethod)
+    if connect_method == 'port':
+        return stem.control.Controller.from_port(port=int(os.environ.get('port', default_torport)))
+    elif connect_method == 'socket':
+        return stem.control.Controller.from_socket_file(path=os.environ.get('socket', default_torsocket))
+    else:
+        print("env.connectmethod contains an invalid value. Please specify either 'port' or 'socket'.", file=sys.stderr)
+        sys.exit(-1)
+
+
+#########################
+# Base Class
+#########################
+
+
+class TorPlugin(object):
+    def __init__(self):
+        raise NotImplementedError
+
+    def conf(self):
+        raise NotImplementedError
+
+    @staticmethod
+    def conf_from_dict(graph, labels):
+        # header
+        for key, val in graph.iteritems():
+            print('graph_{} {}'.format(key, val))
+        # values
+        for label, attributes in labels.iteritems():
+            for key, val in attributes.iteritems():
+                print('{}.{} {}'.format(label, key, val))
+
+    @staticmethod
+    def autoconf():
+        try:
+            with gen_controller() as controller:
+                try:
+                    authenticate(controller)
+                    print('yes')
+                except stem.connection.AuthenticationFailure as e:
+                    print('no (Authentication failed: {})'.format(e))
+
+        except stem.connection:
+            print('no (Connection failed)')
+
+    @staticmethod
+    def suggest():
+        options = ['connections', 'traffic', 'dormant', 'bandwidth', 'flags', 'countries']
+
+        for option in options:
+            print(option)
+
+    def fetch(self):
+        raise NotImplementedError
+
+
+##########################
+# Child Classes
+##########################
+
+
+class TorBandwidth(TorPlugin):
+    def __init__(self):
+        pass
+
+    def conf(self):
+        graph = {'title': 'Observed bandwidth',
+                 'args': '-l 0 --base 1000',
+                 'vlabel': 'bytes/s',
+                 'category': 'Tor',
```

Comment:
`Tor` is not a commonly known category. Please pick one from the [list of well-known categories](http://munin-monitoring.org/wiki/graph_category_list) instead.
I guess, `network` would be suitable, here.